renku-model/src/main/scala/io/renku/graph/model/entities/Plan.scala

```diff
@@ -257,8 +270,11 @@ object StepPlan {
         )
     }
 
+  private val withStrictEntityTypes: Cursor => JsonLDDecoder.Result[Boolean] =
+    _.getEntityTypes.map(types => types.toList.toSet == StepPlan.entityTypes.toList.toSet)
```

Comment:
The `EntityTypes` class has a custom `equals` method so you should be able to do:
```
....map(_ == StepPlan.entityTypes)

Reply:
Ah 🤦🏼 Thank you!